### ISSUE TITLE
static analyzer mode=deep false positive with libwebp/imageio/pnmdec.c

### ISSUE THREAD

**Issue by jzern** [llvm/llvm-project#57515]
`clang-analyzer-core.uninitialized.ArraySubscript in libwebp/imageio/pnmdec.c` at line 118.

Most recently this reproduced with `Debian clang version 14.0.6-2`, but has been observed in earlier versions (`Debian clang version 11.0.1-2` and `clang version 7.0.1 (tags/RELEASE_701/final)`).

```
$ git clone https://chromium.googlesource.com/webm/libwebp
$ cd libwebp
$ git checkout v1.2.4-68-ge8f83de2
$ ./autogen.sh
$ scan-build-14 -o output-deep -analyzer-config mode=deep ./configure --enable-everything --enable-asserts
$ scan-build-14 -o output-deep -analyzer-config mode=deep make -j
```

This looks like a false positive. The core issue is how the for loop is analyzed when using strlen() [1]. To summarize, the analyzer assumes a `data_size` of 1, so we end up with 1 character in `out[]`, which is null terminated [2]. `strlen(out) == 1`, but we see _"The value 1 is assigned to 'i'"_,  _"Assuming the condition is true"_, _"Loop condition is true.  Entering loop body"_. Changing the condition to `out[i] != '\0'` hides the issue.

[1]: https://chromium.googlesource.com/webm/libwebp/+/v1.2.4-68-ge8f83de2/imageio/pnmdec.c#116
[2]: https://chromium.googlesource.com/webm/libwebp/+/v1.2.4-68-ge8f83de2/imageio/pnmdec.c#62
[report-84M9Wv.sarif.gz](https://github.com/llvm/llvm-project/files/9474062/report-84M9Wv.sarif.gz)
[report-OSbFAC.plist.gz](https://github.com/llvm/llvm-project/files/9474063/report-OSbFAC.plist.gz)
